### PR TITLE
An expression may evaluate to Null but an equality test evaluates to UndefinedIdentifier

### LINKED ISSUE #28: An expression may evaluate to `Null` but an equality test evaluates to `UndefinedIdentifier`
Consider the following:

```
let evaluator = Evaluator::new();

let context = value!({
    "foo": null,
});

let expr = "foo".to_string();

let r = evaluator.eval_in_context(&expr, context).unwrap();
```

In the above snippet, `foo` will evaluate to `null`. This is the same behavior that is seen in jexl. If we expand this to an equality test, however, you will see that the binary comparison is not currently possible:

```
...
let expr = "foo == null".to_string();
let r = evaluator.eval_in_context(&expr, context).unwrap();
```
This will produce the following error:
```
thread '...' panicked at 'called `Result::unwrap()` on an `Err` value: UndefinedIdentifier("null")', jexl-eval/src/lib.rs:816:59
```

Equality comparisons should allow for `null` identifiers.

### PR DESCRIPTION
**Pull request by johnsonw**
An expression may evaluate to Null but an equality test evaluates to UndefinedIdentifier.

Fixes #28.

Support `Null` identifiers in expressions and allow binary comparison of expressions when evaluating.

Signed-off-by: Will Johnson <wjohnson@whamcloud.com>

### CODE REVIEW

**Comment by jgrund** [comment]
Hi, any update here? It would be great to check to be able to check for `null`.

**Comment by jgrund** [comment]
Ping. Anything stopping this from landing?

File: jexl-parser/src/lib.rs [hunk 1]
```diff
@@ -136,6 +136,23 @@ mod tests {
         );
     }
 
+    #[test]
+    fn test_dot_op_equality_with_null() {
+        let exp = "foo.bar == Null";
```
Comment (jhugman) [inline jexl-parser/src/lib.rs:141]:
```suggestion
        let exp = "foo.bar == null";
```

File: jexl-parser/src/parser.lalrpop [hunk 2]
```diff
@@ -126,6 +127,11 @@ String: String = {
     <s: r#"'([^'\\]*(\\')?)*'"#> => s[1..s.len() - 1].to_string().replace("\\'", "'"),
 };
 
+Null: Option<Box<Expression>> = {
+    "Null" => None,
+    "null" => None,
+    "NULL" => None,
```
Comment (jhugman) [inline jexl-parser/src/parser.lalrpop:133]: If possible, I'd prefer this to be just `null`:

```suggestion
    "null" => None,
```

**Review by jhugman — APPROVED** [review]
Thank you so much for this PR. I'm just sorry it took so long for someone to get to reviewing it. 

I have added feedback, as commit suggestions: as it's been so long since you worked on this, you may not have the context or desire to make the necessary changes.

I'll make a patch release once you accept the changes, and we can merge them in.

Thank you again!

**Comment by johnsonw** [comment]
Suggestions applied. Thanks for reviewing!

**Comment by johnsonw** [comment]
Thank you for merging! Is there any way a new release could be cut and pushed up to crates.io?

**Comment by jgrund** [comment]
Hi, any update? We'd like to get this new functionality in the next release.

**Pull request merged**